src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,410 @@
+        namespace_name = NEW_CLASS_NAME_TO_NAMESPACES[new_class_name][0]
+        subpackage_name = namespace_name.split(".")[1]
+        return ast.Call(
+            func=ast.Attribute(value=ast.Name(id=subpackage_name), attr=new_class_name),
+            args=[],
+            keywords=[],
+        )
+
+
+class SerdeImportFromPredictorRenamer(Modifier):
+    """A class to update import statements starting with ``from sagemaker.predictor``."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the import statement imports a SerDe from the
+        ``sagemaker.predictor`` module.
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: True if and only if the ``ast.ImportFrom`` imports a SerDe
+                from the ``sagemaker.predictor`` module.
+        """
+        return node.module == "sagemaker.predictor" and any(
+            name.name in (OLD_CLASS_NAMES | OLD_OBJECT_NAMES) for name in node.names
+        )
+
+    def modify_node(self, node):
+        """Removes the imported SerDe classes, as applicable.
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            ast.ImportFrom: a node that represents a import statement, which has
+                been modified to remove imported serializers. If nothing is
+                imported, None is returned.
+        """
+        i = 0
+        while i < len(node.names):
+            name = node.names[i].name
+            if name in OLD_CLASS_NAMES | OLD_OBJECT_NAMES:
+                node.names.pop(i)
+            else:
+                i += 1
+
+        return node if node.names else None
+
+
+class SerdeImportFromAmazonCommonRenamer(Modifier):
+    """A class to update import statements starting with ``from sagemaker.amazon.common``."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the import statement imports a SerDe from the
+        ``sagemaker.amazon.common`` module.
+
+        This checks for:
+        - ``sagemaker.amazon.common.numpy_to_record_serializer``
+        - ``sagemaker.amazon.common.record_deserializer``
+
+        Args:
+            node (ast.ImportFrom): a node that represents a ``from ... import ... `` statement.
+                For more, see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: True if and only if the ``ast.ImportFrom`` imports a SerDe from
+                the ``sagemaker.amazon.common`` module.
+        """
+        return node.module == "sagemaker.amazon.common" and any(
+            [alias.name in OLD_AMAZON_CLASS_NAMES for alias in node.names]
```

Comment:
remove `[` as unneeded as generator expression is fine
```suggestion
            alias.name in OLD_AMAZON_CLASS_NAMES for alias in node.names
```